2.0/nanoserver-1709/kitchensink/Dockerfile

```diff
@@ -0,0 +1,64 @@
+# escape=`
+
+# Installer image
+FROM microsoft/windowsservercore:1709 AS installer-env
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# Retrieve node and git
+ENV NODE_VERSION 6.11.3
+
+RUN Invoke-WebRequest -UseBasicParsing https://nodejs.org/dist/v${env:NODE_VERSION}/node-v${env:NODE_VERSION}-win-x64.zip -outfile node.zip; `
+    Expand-Archive node.zip -DestinationPath nodejs-tmp; `
+    Move-Item nodejs-tmp/node-v${env:NODE_VERSION}-win-x64 nodejs; `
+    Remove-Item -Force node.zip; `
+    Remove-Item -Force nodejs-tmp; `
+    Invoke-WebRequest -UseBasicParsing https://github.com/git-for-windows/git/releases/download/v2.14.1.windows.1/MinGit-2.14.1-64-bit.zip -outfile git.zip; `
+    Expand-Archive git.zip -DestinationPath git; `
+    Remove-Item -Force git.zip
+
+# Retrieve additional runtimes
+RUN ('https://dotnetcli.blob.core.windows.net/dotnet/preview/Binaries/1.0.7/dotnet-win-x64.1.0.7.zip', `
+     'https://dotnetcli.blob.core.windows.net/dotnet/release/1.1.0/Binaries/1.1.4/dotnet-win-x64.1.1.4.zip' ) `
+     | % { `
+        Invoke-WebRequest $_ -UseBasicParsing -OutFile dotnet.zip; `
+        # Ignore errors and don't use -Force. We want to keep the existing dotnet.exe instead of overwriting `
+        Expand-Archive dotnet.zip -DestinationPath dotnet -ErrorAction SilentlyContinue; `
+        Remove-Item -Force dotnet.zip; `
+     }
+
+# Retrieve Docker SDK for when building a solution that has the Docker tools project.
+RUN Invoke-WebRequest -UseBasicParsing https://distaspnet.blob.core.windows.net/sdk/Microsoft.Docker.Sdk.zip -OutFile sdk.zip; `
+    Expand-Archive sdk.zip -DestinationPath sdks; `
+    Remove-Item -Force sdk.zip
+
+
+# Build image
+FROM microsoft/dotnet:2.0.0-sdk-2.0.2-nanoserver-1709
+
+# Set up environment
+ENV ASPNETCORE_URLS http://+:80
+ENV ASPNETCORE_PKG_VERSION 2.0.0
+
+# Install node, bower, and git
+COPY --from=installer-env ["nodejs", "C:\\Program Files\\nodejs"]
+WORKDIR "C:\\Program Files\\nodejs"
+RUN npm install -g gulp bower
+WORKDIR c:\
+COPY --from=installer-env ["git", "C:\\Program Files\\git"]
+RUN setx PATH "%PATH%;C:\Program Files\nodejs;C:\Program Files\git\cmd"
```

Comment:
It might be worth adding a comment to point out that RUN switches from PowerShell syntax to batch. It's subtle, but very important for maintainers to know this.

Reply:
Good point - added comment.